lib/modules/platform/scmm/index.ts

```diff
@@ -0,0 +1,229 @@
+import type {
+  CreatePRConfig,
+  EnsureIssueConfig,
+  FindPRConfig,
+  Issue,
+  PlatformParams,
+  PlatformResult,
+  Pr,
+  RepoParams,
+  RepoResult,
+  UpdatePrConfig,
+} from '../types';
+import { repoFingerprint } from '../util';
+import { smartTruncate } from '../utils/pr-body';
+import { mapPrFromScmToRenovate } from './mapper';
+import ScmClient from './scm-client';
+import { getRepoUrl, matchPrState, smartLinks } from './utils';
+
+interface SCMMRepoConfig {
+  repository: string;
+  prList: Pr[] | null;
+  defaultBranch: string;
+}
+
+export const id = 'scmm';
+
+let config: SCMMRepoConfig = {} as any;
+let scmmClient: ScmClient;
+
+export async function initPlatform({
+  endpoint,
+  token,
+}: PlatformParams): Promise<PlatformResult> {
+  if (!endpoint) {
+    throw new Error('SCM-Manager endpoint not configured');
+  }
+
+  if (!token) {
+    throw new Error('SCM-Manager api token not configured');
+  }
+
+  scmmClient = new ScmClient(endpoint, token);
+
+  const me = await scmmClient.getCurrentUser();
+  const gitAuthor = `${me.displayName ?? me.username} <${me.mail ?? ''}>`;
+  const result = { endpoint, gitAuthor };
+
+  logger.info(`Plattform initialized ${JSON.stringify(result)}`);
+
+  return result;
+}
+
+export async function initRepo({
+  repository,
+  gitUrl,
+}: RepoParams): Promise<RepoResult> {
+  const repo = await scmmClient.getRepo(repository);
+  const defaultBranch = await scmmClient.getDefaultBranch(repo);
+  const url = getRepoUrl(
+    repo,
+    gitUrl,
+    hostRules.find({ hostType: id, url: scmmClient.getEndpoint() }).username ??
+      '',
+    process.env.RENOVATE_TOKEN ?? '',
+  );
+
+  config = {} as any;
+  config.repository = repository;
+  config.defaultBranch = defaultBranch;
+
+  await git.initRepo({
+    ...config,
+    url,
+  });
+
+  // Reset cached resources
+  invalidatePrCache();
+
+  const result = {
+    defaultBranch: config.defaultBranch,
+    isFork: false,
+    repoFingerprint: repoFingerprint(
+      config.repository,
+      scmmClient.getEndpoint()
+    ),
+  };
+
+  logger.info(`Repo initialized: ${JSON.stringify(result)}`);
+
+  return result;
+}
+
+export async function getRepos(): Promise<string[]> {
+  const repos = await scmmClient.getAllRepos();
+  const result = repos.map((repo) => `${repo.namespace}/${repo.name}`);
+  logger.info(`Discoverd ${repos.length} repos`);
+
+  return result;
+}
+
+export async function getBranchPr(branchName: string): Promise<Pr | null> {
+  return await findPr({ branchName, state: 'open' });
+}
+
+export async function findPr({
+  branchName,
+  prTitle,
+  state = 'all',
+}: FindPRConfig): Promise<Pr | null> {
+  const inProgressPrs = await getPrList();
+  const result = inProgressPrs.find(
+    (pr) =>
+      branchName === pr.sourceBranch &&
+      (!prTitle || prTitle === pr.title) &&
+      matchPrState(pr, state)
+  );
+
+  if (result) {
+    logger.info(`Found PR ${JSON.stringify(result)}`);
+    return result;
+  }
+
+  logger.info(
```

Comment:
![Screenshot from 2023-08-07 14-10-14](https://github.com/scm-manager/renovate/assets/50202942/35324221-fe35-47af-b6c9-956acb6cc16b)

This printed too often and confused me. We should remove this or make it log level debug.

Reply:
fixed